router/overlay_switch.go

```diff
@@ -166,6 +170,9 @@ type subForwarderEvent struct {
 
 	// is this an error event?
 	err error
+
+	// event to indicate if forwarder is in healthy state to be considered for best forwarded
+	healthy bool
```

Comment:
I think it would be better to have this as a `*bool`, then you can tell the difference between an event sent with this set to `false` and an event where it was never set.